### ISSUE TITLE
Incorrect path calculation for long path after update to 4.27

### ISSUE THREAD

**Issue by ivasyliev** [ZioYuri78/GraphAStarExample#6]
I was trying to use Engine function FindPathToLocationSynchronously and I found the issue that result with Path points are incorrect after engine update.
At some point engine call the function AGraphAStarNavMesh::FindPath and this function return path points. If path is longer then X cells  (5 in example and 7 in my proj. I think it depends from world distance) then this function return such array
[1, 1, 2, 3, .... x]. So start point is there twice.

**Comment by ZioYuri78** [comment]
That is interesting, 
as you can see in GraphAStarNavMesh.cpp at [line 179](https://github.com/ZioYuri78/GraphAStarExample/blob/940a7a53e59c3f4a07ffa7a2663d094f0f04695f/Source/GraphAStarExample/Private/AI/GraphAStarNavMesh.cpp#L179) i manually add the starting node because FGraphAStar::FindPath doesn't add it, or at least is what is always happened to me but i never tested it with FindPathToLocationSynchronously.
For some reason sometime when you use FindPathToLocationSynchronously the starting node is counted so we have twice of it after my Add call.
I need to go deeper with the debugger to understand why this happen, meanwhile thanks for the heads up!

**Comment by ivasyliev** [comment]
yeah, I read though the code a bit and I've seen your logic. But I didn't found anything obvious, that's why I'm reaching you. I don't know why there is dependency from path length and from engine version. But looks like it was ok in 4.26.
thanks for quick response. Let me know if you'll need any support

**Comment by ZioYuri78** [comment]
I did a small workaround, it not really solve the issue because i still have to figure out what happen but at least it avoid duplication in paths longers than 7 tiles and it let us move of only 1 tile.

**Comment by ZioYuri78** [comment]
Workaround introduced a bug so i had to revert, issue still unresolved.
